### ISSUE TITLE
Evaluating accuracy of model predictions on own annotated test dataset

### ISSUE THREAD

**Issue by HjalteMann** [matterport/Mask_RCNN#1849]
I have trained a model to detect some specific classes and I can run predictions on test images and output images with predictions as masks. But how do I evaluate the accuracy of the model predictions with my manual annotations of the test dataset (not train or val). Is there a function where I can input my ground truth annotations of the test dataset?

**Comment by JavierClearImageAI** [comment]
there is indeed a function, but you would have to compile the model with metrics=['accuracy'] first.

You can use model.evaluate_generator
[evaluate_generator](https://keras.rstudio.com/reference/evaluate_generator.html).

Nevertheless something doesn't work. It doesn't allow me to use metrics=['accuracy'] when compiling the model, just before the training. It raises an error.

**Comment by mangalbhaskar** [comment]
@HjalteMann [compute_ap](https://github.com/matterport/Mask_RCNN/blob/master/mrcnn/utils.py#L715) provides the average precision (AP) per image for a given IoU threshold. And, [compute_ap_range](https://github.com/matterport/Mask_RCNN/blob/master/mrcnn/utils.py#L754) provides the AP per image for 10 different thresholds from 0.5 till 0.95 per image.

Then you can mean this AP over all the images in your evaluation tests dataset to get the mean average precision (mAP)
